src/sinks/tsd-query-log-sink.ts

```diff
@@ -72,22 +72,25 @@ export class TsdQueryLogSink implements tsdDef.Sink {
     private static stringify(metricsEvent:tsdDef.MetricsEvent):string {
         var transformedMetricsEvent = TsdQueryLogSink.transformMetricsEvent(metricsEvent);
 
-        return JSON.stringify(utils.stenofy(transformedMetricsEvent),
+        return JSON.stringify(utils.stenofy(transformedMetricsEvent, transformedMetricsEvent.annotations["_host"]),
             (key, value) => {
                 if (tsdSink.TsdSink.isMetricSample(value)) {
-                    return {
-                        value: (<tsdDef.MetricSample>value).getValue(),
-                        unit: (<tsdDef.MetricSample>value).getUnit()
+                    // TODO(matthayter): implement unit numerators & denominators
```

Comment:
So MAD supports numerators only at this time. However, all the clients and the format support numerator denominator. They should stand-alone. You elected to do 2F in the meantime and I think most of the work for units would be transferable to 2G.